server/src/Application/Services/Course/CourseService.cs

```diff
@@ -40,21 +41,26 @@ public async Task<IReadOnlyCollection<CourseDto>> GetAllCourses()
         return allCourses.Select(course => new CourseDto(course)).ToList();
     }
 
-    public async Task UpdateCourse(int id, UpdateCourseDto dto)
+    public async Task UpdateCourse(int id, UpdateCourseDto dto, string? applicationUserId)
```

Comment:
I suggest creating an interface in Application layer (e.g. `ICurrentUserAccessor`) and implementing (injecting) it in Api layer. Then we don't need to pass the user to all services and just have it injected.

For example: https://stackoverflow.com/questions/36641338/how-to-get-current-user-in-asp-net-core